scipy/optimize/tests/test_optimize.py

```diff
@@ -3125,3 +3125,454 @@ def f(x):
 
     assert_allclose(res.fun, ref.fun)
     assert_allclose(res.x, ref.x)
+
+def test_gh12513_trustregion_exact_infinite_loop():
+    # gh-12513 reported that optimize.minimize might hang when
+    # method='trust-exact', using a new option ``subproblem_maxiter``,
+    # this can be avoided.
+    J = np.array(
+        [
+            -2.53298102e-07,
+            1.76392040e-06,
+            1.74776130e-06,
+            -4.19479903e-10,
+            1.44167498e-05,
+            1.41703911e-08,
+            1.76392030e-06,
+            4.96030153e-06,
+            -2.35771675e-07,
+            -1.68844985e-06,
+            4.29218258e-07,
+            6.65445159e-07,
+            -3.87045830e-08,
+            -3.17236594e-07,
+            -1.21120169e-06,
+            4.59717313e-08,
+            1.67123246e-06,
+            1.46624675e-08,
+            4.22723383e-08,
+        ]
+    )
+    H = np.array(
+        [
+            [
+                3.67335930e01,
+                -2.52334820e02,
+                1.15477558e01,
+                -1.19933725e-03,
+                -2.06408851e03,
+                -2.05821411e00,
+                -2.52334820e02,
+                -6.52076924e02,
+                -2.71362566e-01,
+                -1.98885126e00,
+                1.22085415e00,
+                2.30220713e00,
+                -9.71278532e-02,
+                -5.11210123e-01,
+                -1.00399562e00,
+                1.43319679e-01,
+                6.03815471e00,
+                -6.38719934e-02,
+                1.65623929e-01,
+            ],
+            [
+                -2.52334820e02,
+                1.76757312e03,
```

Comment:
We probably don't want to format like `black` here as it produces a lot of empty space. I would say something along the lines of 
```python
    H = np.array([
        [3.67335930e01, -2.52334820e02, 1.15477558e01, -1.19933725e-03,
         -2.06408851e03, -2.05821411e00, -2.52334820e02,  -6.52076924e02,
         -2.71362566e-01, -1.98885126e00, 1.22085415e00, 2.30220713e00,
         -9.71278532e-02, -5.11210123e-01, -1.00399562e00, 1.43319679e-01,
         6.03815471e00, -6.38719934e-02, 1.65623929e-01],
        [-2.52334820e02, 1.76757312e03, # ...
         # ...
        # ...
        ])
```

Reply:
@lucascolley, fine by me. How do I achieve that? (short of manually editing the lines)

Reply:
I don't know of a way to do it automatically - formatting of big arrays like this has been a big issue for a while and is indeed the main reason why SciPy doesn't have an auto-formatter like `black`.

The easiest way to do this that I know of is probably to revert the most recent commit, change to my suggestion or something similar manually, then add the few other changes that `black` caught manually aswell.

Reply:
Alright. I ended up using an emacs macro to reformat the matrix. Hopefully this looks good enough.

Reply:
that looks good to me and the linter is happy 👍